pkg/codegen/hcl2/model/print_test.go

```diff
@@ -20,9 +20,22 @@ func TestPrintNoTokens(t *testing.T) {
 						Value: cty.True,
 					},
 				},
+				&Attribute{
+					Name: "literal",
+					Value: &TemplateExpression{
+						Parts: []Expression{
+							&LiteralValueExpression{
+								Value: cty.StringVal("foo${bar}"),
```

Comment:
What does "foo$" and "foo${" look like? Do they need/not-need escaping?

Reply:
Ah yeah this might require looking ahead and making sure each '{' has a valid '}' later on, possibly also cases such as `foo${bar${baz}` , if this should be `foo$${bar${baz}` or `foo${bar$${baz}` (though seems unlikely to occur)

Reply:
It looks like only ${ needs handling given https://developer.hashicorp.com/terraform/language/expressions/strings